### ISSUE TITLE
Crash (SEGV) in Producer class

### ISSUE THREAD

**Issue by joaoe** [confluentinc/confluent-kafka-python#1590]
Description
===========
In my project I derived a new class from Producer to either implement a Mock producer for unit tests or a local Producer that just took snapshots of messages to a local cache.

The code looked like this
```
class MyProducer(Producer):
    def __init__(self, config):
        # Don't call super().__init__()
        self._config = config

MyProducer().flush()
```
and then everything crashed.

Looking at the code, the Producer class internally is represented by `Handle` which has a `rk` property. This is set to `NULL` until `__init__` is called, but a lot of the `Handle` code lacks `NULL` checks.

How to reproduce
================
See code example above.

Checklist
=========
Please provide the following information:

 - [ ] confluent-kafka-python and librdkafka version (`confluent_kafka.version()` and `confluent_kafka.libversion()`): master
 - [ ] Apache Kafka broker version:
 - [ ] Client configuration: `{...}`
 - [ ] Operating system:
 - [ ] Provide client logs (with `'debug': '..'` as necessary)
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by joaoe** [comment]
> Is there any specific reason that you don't want to call `super.init()`?

Hi.
Not really. I started with a mock Producer like
```
class MyProducer(Producer):
    def __init__(self, config):
           # ...
```
that crashed so I just changed to
```
class MyProducer:
    __class__ = Producer
    def __init__(self, config):
           # ...
```

Regardless I think we can both agree that the code should not crash with SEGV :) I submitted a PR with many NULL checks that should make everything more stable, also with the AdminClient class.

**Comment by MSeal** [comment]
Yes inheritance shouldn't SEGV and NULL checks should be in place. The PR touches a lot of places so needs a little more careful review, especially given no one reviewed it for 2 years. I'll mark as HIGH priority so we can get the PR cleaned up and merged or this issue closed if unforeseen issues are present therein.

**Comment by k-raina** [comment]
Thanks for raising this issue!
We have fixed this in PR https://github.com/confluentinc/confluent-kafka-python/pull/2122
